packages/wonder-blocks-core/components/clickable-behavior.js

```diff
@@ -269,13 +298,71 @@ export default class ClickableBehavior extends React.Component<
         this.dragging = false;
     }
 
+    async runCallbackAndMaybeNavigate(e: SyntheticEvent<>) {
+        const {
+            onClick = undefined,
+            beforeNav = undefined,
+            safeWithNav = undefined,
+            skipClientNav,
+            history,
+        } = this.props;
+        let shouldNavigate = true;
+
+        if (onClick) {
```

Comment:
It feels like the onClick should happen after the other two callbacks. This way we could do things like track when modals are opened, etc.

Reply:
Sorry, I don't follow.  Could you explain a bit more about how having `onClick` come last would help with this?

Reply:
Right now clickable things can have an onClick or a href. Right now the href is visited after the tracking, whereas the onClick is fired before the tracking. We should ensure this is consistent so that we don't end up with a case where someone puts a redirect in an onClick and is now not being tracked, due to execution order.

Reply:
I see what you're saying.  It feels a bit weird to artificially delay a `onClick` handler.  I thought the purpose of these new props was to move any tracking out of `onClick` so that we didn't have to manually do the `redirect` ourselves.

Reply:
Maybe we could track whether the consumer provided `onClick` handler calls `e.preventDefault()` and use that to not redirect.  That way we'd be preserving more of the default behavior.